responses.py

```diff
@@ -282,6 +290,15 @@ def _on_request(self, adapter, request, **kwargs):
 
         self._calls.add(request, response)
 
+        request_timeout = kwargs.get('timeout')
+        response_timeout = match.get('timeout')
+        if request_timeout is not None and response_timeout is not None:
+            if request_timeout < response_timeout:
+                raise ConnectTimeout()
+
+        if response_timeout:
+            time.sleep(response_timeout)
```

Comment:
I get what you're going for but unit tests should be fast (very fast) and adding sleeps is generally not a good idea.